packages/vite/src/node/optimizer/index.ts

```diff
@@ -1330,63 +1337,69 @@ export async function optimizedDepNeedsInterop(
   return depInfo?.needsInterop
 }
 
-const optimizedDepsCache = new WeakMap<
-  DepOptimizationMetadata,
-  Map<string, string>
->()
-export async function loadOptimizedDep(
-  file: string,
-  depsOptimizer: DepsOptimizer,
-): Promise<string> {
-  const outputFiles = optimizedDepsCache.get(depsOptimizer.metadata)
-  if (outputFiles) {
-    const outputFile = outputFiles.get(file)
-    if (outputFile) return outputFile
-  }
-  return fsp.readFile(file, 'utf-8')
-}
-
-/**
- * Processes that write to the deps cache directory adds a `_writing` lock to
- * inform other processes of so. So before doing any work on it, they can wait
- * for the file to be removed to know it's ready.
- *
- * Returns true if successfully waited for unlock, false if lock timed out.
- */
-async function waitOptimizerWriteLock(depsCacheDir: string, logger: Logger) {
-  const writingPath = path.join(depsCacheDir, '_writing')
-  const tryAgainMs = 100
-
-  // if _writing exist, we wait for a maximum of 500ms before assuming something
-  // is not right
-  let maxWaitTime = 500
-  let waited = 0
-  let filesLength: number
-
-  while (fs.existsSync(writingPath)) {
-    // on the first run, we check the number of files it started with for later use
-    filesLength ??= (await fsp.readdir(depsCacheDir)).length
-
-    await new Promise((r) => setTimeout(r, tryAgainMs))
-    waited += tryAgainMs
-
-    if (waited >= maxWaitTime) {
-      const newFilesLength = (await fsp.readdir(depsCacheDir)).length
-
-      // after 500ms, if the number of files is the same, assume previous process
-      // terminated and didn't cleanup `_writing` lock. clear the directory.
-      if (filesLength === newFilesLength) {
-        logger.info('Outdated deps cache, forcing re-optimization...')
-        await fsp.rm(depsCacheDir, { recursive: true, force: true })
-        return false
-      }
-      // new files were saved, wait a bit longer to decide again.
-      else {
-        maxWaitTime += 500
-        filesLength = newFilesLength
+const MAX_TEMP_DIR_AGE_MS = 24 * 60 * 60 * 1000
+export async function cleanupDepsCacheStaleDirs(
```

Comment:
nice! Maybe add a debug log and include the names of deleted directories. (and improve the match pattern to include the timestamp to reduce risk of removing unrelated dirs)

It has a small risk of wrecking havoc if we ever add another kind of `_temp_` dir here that was unintentionally removed. Testing is never going to catch something removed after a day+, so debug logging (or even visible logging on info/warn?) would help understand that problem.

Reply:
This is the same code we had before https://github.com/vitejs/vite/pull/12582. I prefer we improve it in another PR to avoid making the diff bigger. I'm thinking we should start using `.vite/.temp/deps_hash` instead of `.vite/deps_temp_hash` so we have all the temp files under a single `.temp` and there is no way to get collision.